### PR TITLE
Fix default manual carousel bugs

### PR DESCRIPTION
**Pull request by rymorale**
## Description
- fix default manual carousel index calculation when calculating default indices. the default manual carousel center index is 0 so the left index should be `imageUris.size - 1`.
- use parseCarouselItemsFromString for both constructors

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by prudrabhat — APPROVED** [review]

File: code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/builders/ManualCarouselNotificationBuilder.kt [hunk 1]
```diff
@@ -215,13 +213,14 @@ internal object ManualCarouselNotificationBuilder {
                     )
                 } else {
                     Triple(
+                        imageUris.size - 1,
```
Comment (prudrabhat) [inline code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/builders/ManualCarouselNotificationBuilder.kt:216]: Doe we check if this is >0 in the calling method somewhere?

File: code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/builders/ManualCarouselNotificationBuilder.kt [hunk 2]
```diff
@@ -215,13 +213,14 @@ internal object ManualCarouselNotificationBuilder {
...
                         PushTemplateConstants.DefaultValues.MANUAL_CAROUSEL_START_INDEX,
-                        PushTemplateConstants.DefaultValues.MANUAL_CAROUSEL_START_INDEX + 1,
-                        PushTemplateConstants.DefaultValues.MANUAL_CAROUSEL_START_INDEX + 2
+                        PushTemplateConstants.DefaultValues.MANUAL_CAROUSEL_START_INDEX + 1
                     )
                 }
         }
 
+        pushTemplate.centerImageIndex = carouselIndices.second
```
Comment (prudrabhat) [inline code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/builders/ManualCarouselNotificationBuilder.kt:223]: Can you elaborate on why this is required?
This method ideally should only responsible for calculating the indices. Manipulating the template here will be difficult to maintain later on. Can this operation be made at a different place?

**Review by prudrabhat — CHANGES_REQUESTED** [review]

File: code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/builders/ManualCarouselNotificationBuilder.kt (hunk 1 above)
Comment (rymorale) [reply code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/builders/ManualCarouselNotificationBuilder.kt:216]: we verify that we have at least 3 images before calculating indices so this is guaranteed to be greater than 0.

File: code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/builders/ManualCarouselNotificationBuilder.kt (hunk 2 above)
Comment (rymorale) [reply code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/builders/ManualCarouselNotificationBuilder.kt:223]: yes, i can move this out of the indices calculation.

**Review by prudrabhat — APPROVED** [review]

**Pull request merged**